src/javascripts/Runners/assetHandlers/scriptFragmentHandler.js

```diff
@@ -36,13 +39,16 @@ ngapp.run(function(mergeAssetService, assetHelpers, pexService, progressLogger,
             }));
     };
 
-    let findScriptFragments = function(merge, plugin, folder) {
+    let findScriptFragments = function(merge, plugin, folder, useMergePlugin) {
         let pluginFile = getPluginHandle(merge, plugin),
             fragmentFiles = getFragmentsFromDisk(plugin, folder);
         if (!pluginFile) return fragmentFiles;
-        return fragmentGroups.reduce((fragments, group) => {
-            return getFragmentsFromPlugin(pluginFile, group, fragments);
-        }, []).filter(fragment => {
+        if (!useMergePlugin || !mergedPluginFragments || !mergedPluginFragments.length) {
```

Comment:
`!mergedPluginFragments` will always be false because you initialize it to an empty array on line 11, it can be removed.

Reply:
the way this condition is written, it will set the contents of the `mergedPluginFragments` array whenever `userMergePlugin` is set to false.  The other conditions don't really matter because you're using or.  I don't think this was your intent.

Also, you're going to create bugs by having a `mergedPluginFragments` array in the handler.  The data needs to be stored within the `merge` in order to be tracked properly.  If you need to edit the elements in the merged plugin rather than the elements in the plugin being merged I'd recommend retrieving the fragments in the `handle` function instead of retrieving them here.

Reply:
The first part you mentioned was my intent - this is how it was before in order to generate the information for the data tab before you merge. It could also be saved there I guess, but it's unlikely the user will call it multiple times without making changes that need to be rescanned.

In the actual merge, it only scans the merged plugin once instead, as it's a single plugin rather then multiple.

Reply:
The second part makes sense! 👍